examples/automated-role-assignment/main.bal

```diff
@@ -14,85 +14,109 @@
 // specific language governing permissions and limitations
 // under the License.
 
+import ballerina/log;
+import ballerina/mime;
+import ballerina/task;
+import ballerinax/discord;
 import ballerina/http;
-import ballerina/io;
 
-public function main() returns error? {
-    check sendRoleAssignmentMessage();
-    check addReactionsToMessage();
-    check startRoleAssignmentListener();
-}
+const string CHANNEL_ID = "YOUR_CHANNEL_ID";
+const string GUILD_ID = "YOUR_GUILD_ID";
 
-function sendRoleAssignmentMessage() returns error? {
-    http:Client discord = check new ("https://discord.com/api/v9");
+configurable string clientId = ?;
+configurable string clientSecret = ?;
+configurable string[] scopes = ?;
+
+discord:ConnectionConfig discordConfig = {
+    auth: {
+        clientId,
+        clientSecret,
+        scopes
+    }
+};
+final discord:Client discord = check new (discordConfig);
 
-    json message = {
-        "content": "React to this message with the following emojis to get your roles:\n\n🔴 - Role1\n🔵 - Role2\n🟢 - Role3"
+public function main() returns error? {
+    discord:Create_messageHeaders headers = {
+        Content\-Type: mime:APPLICATION_FORM_URLENCODED
+    };
+    discord:channel_id_messages_body payload = {
+        content: "React to this message with the following emojis to get your roles:\n\n🔴 - Role1\n🔵 - Role2\n🟢 - Role3"
     };
 
-    http:Request request = new;
-    request.setPayload(message);
-    request.addHeader("Authorization", "Bot BOT_TOKEN");
-    request.addHeader("Content-Type", "application/json");
+    discord:MessageResponse|error createMessageResponse = discord->/channels/[CHANNEL_ID]/messages.post(headers, payload);
+    if createMessageResponse is error {
+        log:printError("Error creating a message: ", createMessageResponse);
+        return;
+    }
 
-    http:Response response = check discord->post("/channels/762316177417961504/messages", request);
-    io:println("Role Assignment Message Sent: ", response.getJsonPayload());
+    task:JobId _ = check task:scheduleJobRecurByFrequency(new RoleAssignmentJob(createMessageResponse.id), 60 * 60 * 24);
```

Comment:
I'm just thinking whether we should keep assigning and changing the roles periodically or, should it be just a one time thing (say we give them one day and assign the roles after that based on the reactions). wdyt?

Reply:
+1

Reply:
Fixed with [97644c0be36988478a3293c5385994f8e8c5e1a4](https://github.com/KamalikaSene/module-ballerinax-discord/pull/3/commits/97644c0be36988478a3293c5385994f8e8c5e1a4)